redhat-developer/red-hat-developers-documentation-rhdh
# chore: Added basic Vale configuration to validate style guide

With this configuration, the user can use Vale CLI to test that in AsciiDoc documents:

- Language is compliant with the Red Hat style guide.
- AsciiDoc attributes replace "Backstage", "Developer Hub", and "Red Hat Developer Hub".

It does not automate the use of Vale in GitHub actions.

To test the local rules:

1. Install Vale. See https://redhat-documentation.github.io/vale-at-red-hat/docs/main/user-guide/installing-vale-cli/
2. Run the tests: 

```
$ vale .vale-styles/DeveloperHub/tests/ --config .vale-styles/DeveloperHub/tests/.vale.ini 

 .vale-styles/DeveloperHub/tests/fail.adoc
 1:11  error  Use the AsciiDoc attribute      DeveloperHub.Attributes 
              '{product}' rather than                                 
              '{product} Hub'.                                        
 2:9   error  Use the AsciiDoc attribute      DeveloperHub.Attributes 
              '{product}' rather than                                 
              'product Hub'.                                          
 3:11  error  Use the AsciiDoc attribute      DeveloperHub.Attributes 
              '{product}' rather than                                 
              'Backstage'.                                            
 4:9   error  Use the AsciiDoc attribute      DeveloperHub.Attributes 
              '{product-short}' rather than                           
              'Developer Hub'.                                        
 5:11  error  Use the AsciiDoc attribute      DeveloperHub.Attributes 
              '{product}' rather than 'Red                            
              Hat Developer Hub'.                                     
 6:9   error  Use the AsciiDoc attribute      DeveloperHub.Attributes 
              '{product}' rather than 'Red                            
              Hat Developer Hub'.                                     
 7:11  error  Use the AsciiDoc attribute      DeveloperHub.Attributes 
              '{product}' rather than 'Red                            
              Hat {product}'.                                         

✖ 7 errors, 0 warnings and 0 suggestions in 2 files.
```
